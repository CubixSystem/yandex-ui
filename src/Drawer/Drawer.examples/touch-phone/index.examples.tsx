import { EXAMPLE_TOKEN, parameters } from '../examples-config';

export * from './big.examples';
export * from './custom-title.examples';
export * from './nested.examples';
export * from './small.examples';
export * from './playground.examples';

export default {
    title: EXAMPLE_TOKEN,
    parameters,
};
