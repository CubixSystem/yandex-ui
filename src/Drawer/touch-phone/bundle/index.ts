import { compose } from '@bem-react/core';

// base
import { Drawer as DrawerTouchPhone } from '../../Drawer@touch-phone';
// _view
import { withViewDefault } from '../../_view/Drawer_view_default';

export * from '../../Drawer@touch-phone';

export const Drawer = compose(withViewDefault)(DrawerTouchPhone);
