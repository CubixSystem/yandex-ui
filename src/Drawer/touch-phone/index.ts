export * from '../Drawer@touch-phone';
export * from '../_view/Drawer_view_default';
