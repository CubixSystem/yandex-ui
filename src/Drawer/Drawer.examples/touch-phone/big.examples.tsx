import React, { useState } from 'react';
import { withKnobs, select, boolean, number, text } from '@storybook/addon-knobs';

import { EXAMPLE_TOKEN, createDecorators, parameters } from '../examples-config';
import { Drawer } from '../../touch-phone/bundle';

import '../examples-styles.css';

export default {
    title: EXAMPLE_TOKEN,
    decorators: [withKnobs, ...createDecorators({ scope: 'touch-phone' })],
    parameters,
};

export const Big = () => {
    const scopeRef = React.useRef<HTMLDivElement>(null);
    const [visible, setVisible] = useState(false);

    const maxSize = text('maxSize', '');
    const nested = boolean('nested', false);
    const dragDisabled = boolean('dragDisabled', false);
    const direction = select('direction', ['left', 'right', 'bottom'], 'bottom');
    const animation = {
        tension: number('tension', 230),
        friction: number('friction', 24),
        disabled: boolean('disable animation', false),
        dragImmediate: boolean('drag immediate', false),
    };

    return (
        <div ref={scopeRef}>
            <p>
                Phasellus sollicitudin in pellentesque cras sagittis platea mattis himenaeos, dui ligula congue ad nisi
                tempor laoreet lacus, etiam gravida taciti mauris adipiscing id erat. Mollis scelerisque vivamus sit
                dictum ultrices eros, suscipit varius cursus litora lectus montes, et posuere diam mauris conubia.
                Inceptos metus vel ac hendrerit lacinia arcu taciti potenti, vulputate class mollis maecenas orci nibh.
            </p>
            <p>
                Phasellus sollicitudin in pellentesque cras sagittis platea mattis himenaeos, dui ligula congue ad nisi
                tempor laoreet lacus, etiam gravida taciti mauris adipiscing id erat. Mollis scelerisque vivamus sit
                dictum ultrices eros, suscipit varius cursus litora lectus montes, et posuere diam mauris conubia.
                Inceptos metus vel ac hendrerit lacinia arcu taciti potenti, vulputate class mollis maecenas orci nibh.
            </p>
            <p>
                <button onClick={() => setVisible(true)} data-testid="opener">
                    Открыть большую шторку
                </button>
            </p>
            <p>
                Phasellus sollicitudin in pellentesque cras sagittis platea mattis himenaeos, dui ligula congue ad nisi
                tempor laoreet lacus, etiam gravida taciti mauris adipiscing id erat. Mollis scelerisque vivamus sit
                dictum ultrices eros, suscipit varius cursus litora lectus montes, et posuere diam mauris conubia.
                Inceptos metus vel ac hendrerit lacinia arcu taciti potenti, vulputate class mollis maecenas orci nibh.
            </p>
            <p>
                Phasellus sollicitudin in pellentesque cras sagittis platea mattis himenaeos, dui ligula congue ad nisi
                tempor laoreet lacus, etiam gravida taciti mauris adipiscing id erat. Mollis scelerisque vivamus sit
                dictum ultrices eros, suscipit varius cursus litora lectus montes, et posuere diam mauris conubia.
                Inceptos metus vel ac hendrerit lacinia arcu taciti potenti, vulputate class mollis maecenas orci nibh.
            </p>
            <p>
                Phasellus sollicitudin in pellentesque cras sagittis platea mattis himenaeos, dui ligula congue ad nisi
                tempor laoreet lacus, etiam gravida taciti mauris adipiscing id erat. Mollis scelerisque vivamus sit
                dictum ultrices eros, suscipit varius cursus litora lectus montes, et posuere diam mauris conubia.
                Inceptos metus vel ac hendrerit lacinia arcu taciti potenti, vulputate class mollis maecenas orci nibh.
            </p>
            <Drawer
                visible={visible}
                onClose={() => setVisible(false)}
                scope={scopeRef}
                nested={nested}
                dragDisabled={dragDisabled}
                direction={direction}
                animation={animation}
                maxSize={maxSize}
                view="default"
            >
                <div className="DrawerInnerContent">
                    <p>
                        Phasellus sollicitudin in pellentesque cras sagittis platea mattis himenaeos, dui ligula congue
                        ad nisi tempor laoreet lacus, etiam gravida taciti mauris adipiscing id erat. Mollis scelerisque
                        vivamus sit dictum ultrices eros, suscipit varius cursus litora lectus montes, et posuere diam
                        mauris conubia. Inceptos metus vel ac hendrerit lacinia arcu taciti potenti, vulputate class
                        mollis maecenas orci nibh.
                    </p>
                    <p>
                        Phasellus sollicitudin in pellentesque cras sagittis platea mattis himenaeos, dui ligula congue
                        ad nisi tempor laoreet lacus, etiam gravida taciti mauris adipiscing id erat. Mollis scelerisque
                        vivamus sit dictum ultrices eros, suscipit varius cursus litora lectus montes, et posuere diam
                        mauris conubia. Inceptos metus vel ac hendrerit lacinia arcu taciti potenti, vulputate class
                        mollis maecenas orci nibh.
                    </p>
                    <p>
                        Phasellus sollicitudin in pellentesque cras sagittis platea mattis himenaeos, dui ligula congue
                        ad nisi tempor laoreet lacus, etiam gravida taciti mauris adipiscing id erat. Mollis scelerisque
                        vivamus sit dictum ultrices eros, suscipit varius cursus litora lectus montes, et posuere diam
                        mauris conubia. Inceptos metus vel ac hendrerit lacinia arcu taciti potenti, vulputate class
                        mollis maecenas orci nibh.
                    </p>
                    <p>
                        Phasellus sollicitudin in pellentesque cras sagittis platea mattis himenaeos, dui ligula congue
                        ad nisi tempor laoreet lacus, etiam gravida taciti mauris adipiscing id erat. Mollis scelerisque
                        vivamus sit dictum ultrices eros, suscipit varius cursus litora lectus montes, et posuere diam
                        mauris conubia. Inceptos metus vel ac hendrerit lacinia arcu taciti potenti, vulputate class
                        mollis maecenas orci nibh.
                    </p>
                    <button onClick={() => setVisible(false)}>Закрыть шторку</button>
                </div>
            </Drawer>
        </div>
    );
};

Big.story = {
    name: 'big',
};
