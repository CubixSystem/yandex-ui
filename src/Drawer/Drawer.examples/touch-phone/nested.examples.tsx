import React, { useState } from 'react';
import { boolean, number, withKnobs } from '@storybook/addon-knobs';

import { EXAMPLE_TOKEN, createDecorators, parameters } from '../examples-config';
import { Drawer } from '../../touch-phone/bundle';

import '../examples-styles.css';

export default {
    title: EXAMPLE_TOKEN,
    decorators: [withKnobs, ...createDecorators({ scope: 'touch-phone' })],
    parameters,
};

export const Nested = () => {
    const scopeRef = React.useRef<HTMLDivElement>(null);
    const [visible, setVisible] = useState(false);
    const [nestedVisible, setNestedVisible] = useState(false);

    const dragDisabled = boolean('dragDisabled', false);
    const animation = {
        tension: number('tension', 230),
        friction: number('friction', 24),
        disabled: boolean('disable animation', false),
        dragImmediate: boolean('drag immediate', false),
    };

    return (
        <div ref={scopeRef}>
            <p>
                Phasellus sollicitudin in pellentesque cras sagittis platea mattis himenaeos, dui ligula congue ad nisi
                tempor laoreet lacus, etiam gravida taciti mauris adipiscing id erat. Mollis scelerisque vivamus sit
                dictum ultrices eros, suscipit varius cursus litora lectus montes, et posuere diam mauris conubia.
                Inceptos metus vel ac hendrerit lacinia arcu taciti potenti, vulputate class mollis maecenas orci nibh.
            </p>
            <p>
                Phasellus sollicitudin in pellentesque cras sagittis platea mattis himenaeos, dui ligula congue ad nisi
                tempor laoreet lacus, etiam gravida taciti mauris adipiscing id erat. Mollis scelerisque vivamus sit
                dictum ultrices eros, suscipit varius cursus litora lectus montes, et posuere diam mauris conubia.
                Inceptos metus vel ac hendrerit lacinia arcu taciti potenti, vulputate class mollis maecenas orci nibh.
            </p>
            <p>
                <button onClick={() => setVisible(true)} data-testid="opener-primary">
                    Открыть шторку
                </button>
            </p>
            <p>
                Phasellus sollicitudin in pellentesque cras sagittis platea mattis himenaeos, dui ligula congue ad nisi
                tempor laoreet lacus, etiam gravida taciti mauris adipiscing id erat. Mollis scelerisque vivamus sit
                dictum ultrices eros, suscipit varius cursus litora lectus montes, et posuere diam mauris conubia.
                Inceptos metus vel ac hendrerit lacinia arcu taciti potenti, vulputate class mollis maecenas orci nibh.
            </p>
            <p>
                Phasellus sollicitudin in pellentesque cras sagittis platea mattis himenaeos, dui ligula congue ad nisi
                tempor laoreet lacus, etiam gravida taciti mauris adipiscing id erat. Mollis scelerisque vivamus sit
                dictum ultrices eros, suscipit varius cursus litora lectus montes, et posuere diam mauris conubia.
                Inceptos metus vel ac hendrerit lacinia arcu taciti potenti, vulputate class mollis maecenas orci nibh.
            </p>
            <p>
                Phasellus sollicitudin in pellentesque cras sagittis platea mattis himenaeos, dui ligula congue ad nisi
                tempor laoreet lacus, etiam gravida taciti mauris adipiscing id erat. Mollis scelerisque vivamus sit
                dictum ultrices eros, suscipit varius cursus litora lectus montes, et posuere diam mauris conubia.
                Inceptos metus vel ac hendrerit lacinia arcu taciti potenti, vulputate class mollis maecenas orci nibh.
            </p>
            <Drawer
                visible={visible}
                onClose={() => setVisible(false)}
                scope={scopeRef}
                view="default"
                animation={animation}
                dragDisabled={dragDisabled}
            >
                <div className="DrawerInnerContent">
                    <p>
                        Phasellus sollicitudin in pellentesque cras sagittis platea mattis himenaeos, dui ligula congue
                        ad nisi tempor laoreet lacus, etiam gravida taciti mauris adipiscing id erat. Mollis scelerisque
                        vivamus sit dictum ultrices eros, suscipit varius cursus litora lectus montes, et posuere diam
                        mauris conubia. Inceptos metus vel ac hendrerit lacinia arcu taciti potenti, vulputate class
                        mollis maecenas orci nibh.
                    </p>
                    <p>
                        Phasellus sollicitudin in pellentesque cras sagittis platea mattis himenaeos, dui ligula congue
                        ad nisi tempor laoreet lacus, etiam gravida taciti mauris adipiscing id erat. Mollis scelerisque
                        vivamus sit dictum ultrices eros, suscipit varius cursus litora lectus montes, et posuere diam
                        mauris conubia. Inceptos metus vel ac hendrerit lacinia arcu taciti potenti, vulputate class
                        mollis maecenas orci nibh.
                    </p>
                    <p>
                        Phasellus sollicitudin in pellentesque cras sagittis platea mattis himenaeos, dui ligula congue
                        ad nisi tempor laoreet lacus, etiam gravida taciti mauris adipiscing id erat. Mollis scelerisque
                        vivamus sit dictum ultrices eros, suscipit varius cursus litora lectus montes, et posuere diam
                        mauris conubia. Inceptos metus vel ac hendrerit lacinia arcu taciti potenti, vulputate class
                        mollis maecenas orci nibh.
                    </p>
                    <button onClick={() => setVisible(false)}>Закрыть шторку</button>
                    <button onClick={() => setNestedVisible(true)} data-testid="opener-secondary">
                        Отрыть вложенную шторку
                    </button>
                </div>
            </Drawer>
            <Drawer
                visible={nestedVisible}
                onClose={() => setNestedVisible(false)}
                scope={scopeRef}
                animation={animation}
                dragDisabled={dragDisabled}
                view="default"
                nested
            >
                <div className="DrawerInnerContent">
                    <p>
                        Phasellus sollicitudin in pellentesque cras sagittis platea mattis himenaeos, dui ligula congue
                        ad nisi tempor laoreet lacus, etiam gravida taciti mauris adipiscing id erat. Mollis scelerisque
                        vivamus sit dictum ultrices eros, suscipit varius cursus litora lectus montes, et posuere diam
                        mauris conubia. Inceptos metus vel ac hendrerit lacinia arcu taciti potenti, vulputate class
                        mollis maecenas orci nibh.
                    </p>
                    <button onClick={() => setNestedVisible(false)}>Закрыть вложенную шторку</button>
                </div>
            </Drawer>
        </div>
    );
};

Nested.story = {
    name: 'nested',
};
