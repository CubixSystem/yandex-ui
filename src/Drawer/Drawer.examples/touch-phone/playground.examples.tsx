import React, { useState } from 'react';
import { select, boolean, number, text } from '@storybook/addon-knobs';
import { Drawer } from '@yandex-lego/components/Drawer/touch-phone/bundle';

export const Playground = () => {
    const scopeRef = React.useRef<HTMLDivElement>(null);
    const [visible, setVisible] = useState(false);

    const maxSize = text('maxSize', '');
    const nested = boolean('nested', false);
    const dragDisabled = boolean('dragDisabled', false);
    const direction = select('direction', ['left', 'right', 'bottom'], 'bottom');
    const animation = {
        tension: number('tension', 230),
        friction: number('friction', 24),
        disabled: boolean('disable animation', false),
        dragImmediate: boolean('drag immediate', false),
    };

    return (
        <div ref={scopeRef}>
            <p>
                Phasellus sollicitudin in pellentesque cras sagittis platea mattis himenaeos, dui ligula congue ad nisi
                tempor laoreet lacus, etiam gravida taciti mauris adipiscing id erat. Mollis scelerisque vivamus sit
                dictum ultrices eros, suscipit varius cursus litora lectus montes, et posuere diam mauris conubia.
                Inceptos metus vel ac hendrerit lacinia arcu taciti potenti, vulputate class mollis maecenas orci nibh.
            </p>
            <p>
                Phasellus sollicitudin in pellentesque cras sagittis platea mattis himenaeos, dui ligula congue ad nisi
                tempor laoreet lacus, etiam gravida taciti mauris adipiscing id erat. Mollis scelerisque vivamus sit
                dictum ultrices eros, suscipit varius cursus litora lectus montes, et posuere diam mauris conubia.
                Inceptos metus vel ac hendrerit lacinia arcu taciti potenti, vulputate class mollis maecenas orci nibh.
            </p>
            <p>
                <button onClick={() => setVisible(true)} data-testid="opener">
                    Открыть большую шторку
                </button>
            </p>
            <Drawer
                visible={visible}
                onClose={() => setVisible(false)}
                scope={scopeRef}
                nested={nested}
                dragDisabled={dragDisabled}
                direction={direction}
                animation={animation}
                maxSize={maxSize}
                view="default"
            >
                <div className="DrawerInnerContent" style={{ padding: '20px' }}>
                    <p>
                        Phasellus sollicitudin in pellentesque cras sagittis platea mattis himenaeos, dui ligula congue
                        ad nisi tempor laoreet lacus, etiam gravida taciti mauris adipiscing id erat. Mollis scelerisque
                        vivamus sit dictum ultrices eros, suscipit varius cursus litora lectus montes, et posuere diam
                        mauris conubia. Inceptos metus vel ac hendrerit lacinia arcu taciti potenti, vulputate class
                        mollis maecenas orci nibh.
                    </p>
                    <p>
                        Phasellus sollicitudin in pellentesque cras sagittis platea mattis himenaeos, dui ligula congue
                        ad nisi tempor laoreet lacus, etiam gravida taciti mauris adipiscing id erat. Mollis scelerisque
                        vivamus sit dictum ultrices eros, suscipit varius cursus litora lectus montes, et posuere diam
                        mauris conubia. Inceptos metus vel ac hendrerit lacinia arcu taciti potenti, vulputate class
                        mollis maecenas orci nibh.
                    </p>
                    <p>
                        Phasellus sollicitudin in pellentesque cras sagittis platea mattis himenaeos, dui ligula congue
                        ad nisi tempor laoreet lacus, etiam gravida taciti mauris adipiscing id erat. Mollis scelerisque
                        vivamus sit dictum ultrices eros, suscipit varius cursus litora lectus montes, et posuere diam
                        mauris conubia. Inceptos metus vel ac hendrerit lacinia arcu taciti potenti, vulputate class
                        mollis maecenas orci nibh.
                    </p>
                    <p>
                        Phasellus sollicitudin in pellentesque cras sagittis platea mattis himenaeos, dui ligula congue
                        ad nisi tempor laoreet lacus, etiam gravida taciti mauris adipiscing id erat. Mollis scelerisque
                        vivamus sit dictum ultrices eros, suscipit varius cursus litora lectus montes, et posuere diam
                        mauris conubia. Inceptos metus vel ac hendrerit lacinia arcu taciti potenti, vulputate class
                        mollis maecenas orci nibh.
                    </p>
                    <button onClick={() => setVisible(false)}>Закрыть шторку</button>
                </div>
            </Drawer>
        </div>
    );
};
