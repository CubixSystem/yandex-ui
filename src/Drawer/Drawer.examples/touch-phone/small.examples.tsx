import React, { useState } from 'react';
import { withKnobs, select, boolean, number } from '@storybook/addon-knobs';

import { EXAMPLE_TOKEN, createDecorators, parameters } from '../examples-config';
import { Drawer } from '../../touch-phone/bundle';

import '../examples-styles.css';

export default {
    title: EXAMPLE_TOKEN,
    decorators: [withKnobs, ...createDecorators({ scope: 'touch-phone' })],
    parameters,
};

export const Small = () => {
    const scopeRef = React.useRef<HTMLDivElement>(null);
    const [visible, setVisible] = useState(false);

    const nested = boolean('nested', false);
    const dragDisabled = boolean('dragDisabled', false);
    const direction = select('direction', ['left', 'right', 'bottom'], 'bottom');
    const animation = {
        tension: number('tension', 230),
        friction: number('friction', 24),
        disabled: boolean('disable animation', false),
        dragImmediate: boolean('drag immediate', false),
    };

    return (
        <div ref={scopeRef}>
            <p>
                Phasellus sollicitudin in pellentesque cras sagittis platea mattis himenaeos, dui ligula congue ad nisi
                tempor laoreet lacus, etiam gravida taciti mauris adipiscing id erat. Mollis scelerisque vivamus sit
                dictum ultrices eros, suscipit varius cursus litora lectus montes, et posuere diam mauris conubia.
                Inceptos metus vel ac hendrerit lacinia arcu taciti potenti, vulputate class mollis maecenas orci nibh.
            </p>
            <p>
                Phasellus sollicitudin in pellentesque cras sagittis platea mattis himenaeos, dui ligula congue ad nisi
                tempor laoreet lacus, etiam gravida taciti mauris adipiscing id erat. Mollis scelerisque vivamus sit
                dictum ultrices eros, suscipit varius cursus litora lectus montes, et posuere diam mauris conubia.
                Inceptos metus vel ac hendrerit lacinia arcu taciti potenti, vulputate class mollis maecenas orci nibh.
            </p>
            <p>
                <button onClick={() => setVisible(true)}>Открыть маленькую шторку</button>
            </p>
            <p>
                Phasellus sollicitudin in pellentesque cras sagittis platea mattis himenaeos, dui ligula congue ad nisi
                tempor laoreet lacus, etiam gravida taciti mauris adipiscing id erat. Mollis scelerisque vivamus sit
                dictum ultrices eros, suscipit varius cursus litora lectus montes, et posuere diam mauris conubia.
                Inceptos metus vel ac hendrerit lacinia arcu taciti potenti, vulputate class mollis maecenas orci nibh.
            </p>
            <p>
                Phasellus sollicitudin in pellentesque cras sagittis platea mattis himenaeos, dui ligula congue ad nisi
                tempor laoreet lacus, etiam gravida taciti mauris adipiscing id erat. Mollis scelerisque vivamus sit
                dictum ultrices eros, suscipit varius cursus litora lectus montes, et posuere diam mauris conubia.
                Inceptos metus vel ac hendrerit lacinia arcu taciti potenti, vulputate class mollis maecenas orci nibh.
            </p>
            <p>
                Phasellus sollicitudin in pellentesque cras sagittis platea mattis himenaeos, dui ligula congue ad nisi
                tempor laoreet lacus, etiam gravida taciti mauris adipiscing id erat. Mollis scelerisque vivamus sit
                dictum ultrices eros, suscipit varius cursus litora lectus montes, et posuere diam mauris conubia.
                Inceptos metus vel ac hendrerit lacinia arcu taciti potenti, vulputate class mollis maecenas orci nibh.
            </p>
            <Drawer
                visible={visible}
                onClose={() => setVisible(false)}
                scope={scopeRef}
                animation={animation}
                nested={nested}
                dragDisabled={dragDisabled}
                direction={direction}
                view="default"
            >
                <div className="DrawerInnerContent">
                    <p>
                        Phasellus sollicitudin in pellentesque cras sagittis platea mattis himenaeos, dui ligula congue
                        ad nisi tempor laoreet lacus.
                    </p>
                    <button onClick={() => setVisible(false)}>Закрыть шторку</button>
                </div>
            </Drawer>
        </div>
    );
};

Small.story = {
    name: 'small',
};
