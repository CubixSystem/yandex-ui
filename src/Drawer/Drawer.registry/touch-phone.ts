import { Registry } from '@bem-react/di';

import { Popup } from '../../Popup/touch-phone';
import { cnDrawer } from '../Drawer.const';

export const DrawerRegistry = new Registry({ id: cnDrawer() });

DrawerRegistry.set('Popup', Popup);
