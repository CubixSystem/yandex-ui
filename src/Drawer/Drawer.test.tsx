import React, { createRef } from 'react';
import { ReactWrapper, mount, render } from 'enzyme';

import { Drawer } from './touch-phone/bundle';

import { serverEnvironmentSetup, delay } from '../internal/utils';
import { Nullable } from '../typings/utility-types';

describe('Drawer', () => {
    let wrapper: Nullable<ReactWrapper> = null;

    afterEach(() => {
        if (wrapper !== null && wrapper.length > 0) {
            wrapper.unmount();
        }
    });

    describe('server environment', () => {
        serverEnvironmentSetup();

        test('должен вернуть полный шаблон компонента (snapshot)', () => {
            expect(
                render(
                    <Drawer animation={{ tension: 230, friction: 24 }}>
                        <p>Phasellus sollicitudin in pellentesque cras sagittis platea mat</p>
                    </Drawer>,
                ),
            ).toMatchSnapshot();
        });
    });

    describe('client environment', () => {
        test('должен вернуть полный шаблон компонента (snapshot)', () => {
            wrapper = mount(
                <Drawer animation={{ tension: 230, friction: 24 }}>
                    <p>Phasellus sollicitudin in pellentesque cras sagittis platea mat</p>
                </Drawer>,
            );
            expect(wrapper).toMatchSnapshot();
        });

        // eslint-disable-next-line mocha/no-skipped-tests
        test.skip('должен устанавливать ссылку на корневой DOM элемент', async () => {
            const innerRef = createRef<HTMLDivElement>();
            wrapper = mount(<Drawer visible innerRef={innerRef} animation={{ tension: 230, friction: 24 }} />);

            await delay(100);
            expect(innerRef.current).not.toBe(null);
        });
    });
});
