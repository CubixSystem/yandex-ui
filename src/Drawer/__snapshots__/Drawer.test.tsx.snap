// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer client environment должен вернуть полный шаблон компонента (snapshot) 1`] = `
<SimpleComposeWrapper [view:default]
  animation={
    Object {
      "friction": 24,
      "tension": 230,
    }
  }
>
  <RegistryResolver(Drawer)
    animation={
      Object {
        "friction": 24,
        "tension": 230,
      }
    }
    className="Drawer"
  >
    <Drawer
      animation={
        Object {
          "friction": 24,
          "tension": 230,
        }
      }
      className="Drawer"
    >
      <RegistryResolver(Popup2)
        className="Drawer Drawer_direction_bottom"
        style={
          Object {
            "height": "768px",
          }
        }
        visible={false}
      >
        <Popup2
          className="Drawer Drawer_direction_bottom"
          style={
            Object {
              "height": "768px",
            }
          }
          visible={false}
        />
      </RegistryResolver(Popup2)>
    </Drawer>
  </RegistryResolver(Drawer)>
</SimpleComposeWrapper [view:default]>
`;

exports[`Drawer server environment должен вернуть полный шаблон компонента (snapshot) 1`] = `null`;
